src/policy_engine/opa/mod.rs

```diff
@@ -85,6 +97,21 @@ impl PolicyEngine for OPA {
 
         Ok((res_kv["allow"].as_bool().unwrap_or(false), res))
     }
+
+    async fn set_policy(&mut self, input: SetPolicyInput) -> Result<()> {
+        if input.r#type != "opa" {
```

Comment:
Why do we need the `type` in a `SetPolicyInput` here? Does it mean a running attestation-service instance has multiple PolicyEngines escorting?

Reply:
This is to avoid setting the policy type of other engines to OPA, as the source of this policy is the remote caller. Therefore, I believe it is necessary to add a type field here to prevent such errors from occurring.